TestSwitchApi/TestSwitchApi.Tests/CandidatesListResponseTest.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NUnit.Framework;
+using TestSwitchApi.Controllers;
+using TestSwitchApi.Models.DataModels;
+using TestSwitchApi.Models.Request;
+using TestSwitchApi.Models.Response;
+using TestSwitchApi.Repositories;
+
+namespace TestSwitchApi.Tests
+{
+    public class CandidatesListResponseTest
+    {
+        public CandidatesListResponseTest()
+        {
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+        }
+
+        [Test]
+        public void TestingForNoCandidates()
+        {
+            var candidates = new List<CandidateDataModel>();
+            var pageRequest = new PageRequest
+            {
+                Page = 1,
+                PageSize = 10,
+            };
+            var candidatesListResponse = CandidateListResponse.Create(pageRequest, candidates, 0);
+            candidatesListResponse.TotalNumberOfItems.Should().Be(0);
+            candidatesListResponse.Items.Count().Should().Be(0);
+            candidatesListResponse.Page.Should().Be(1);
+            candidatesListResponse.PageSize.Should().Be(10);
+            candidatesListResponse.PreviousPage.Should().BeNullOrEmpty();
+            candidatesListResponse.NextPage.Should().BeNullOrEmpty();
```

Comment:
I think these tests are showing up a problem with CandidateListResponse. Why do you need to specify how many items are in the response when you're also giving the response the list of items itself?
If you look carefully, most of the logic you're testing is actually being done by the test, rather than `CandidateListResponse.Create()`.
The only logic that's actually being done inside the method is setting the values of NextPage and PreviousPage. The rest of these values are being set by the test.
Really you want a method in the app which just takes in all the candidates which exist and a page request object, then decides itself how many of those to include in the response.

Reply:
I think this is partially my fault - for trying to do something clever! :) 

We've often had 'list/search' endpoints that return paginated data.
I wanted to make it easy for the react apps etc to consume this data, so thought it would be clever to wrap it in an object.

The object can then work out for itself if there is a next page, and what the URL for that would be etc etc etc.

The front end can then just call 'nextPage' and everything is lovely and simple.


But the wrapper object itself is much less tidy than I was hoping :(  
Mostly because it needs quite a lot of data
- the base path
- the current query string (or all the properties to rebuild it from)
- the items in that page
- the total number of items in the DB (so it can know if this is the last page)

so it all got a bit messy!
I think the code we are testing here is heavily based on what I did before - which may or may not be a good pattern to follow! 😆 

> Really you want a method in the app which just takes in all the candidates which exist and a page request object, then decides itself how many of those to include in the response.

this exists already! :)  All done in the LINQ for the DB query.

Reply:
The logic has been simplified. I have removed the base class , now there is only one class that works the Next , Prev logic